### ISSUE TITLE
Segmentation fault while mq is open

### ISSUE THREAD

**Issue by jkuba** [mscdex/pmq#5]
While a mq is opened and waiting for an incoming message, I am seeing segmentation faults in node when performing other I/O operations.

I have modified one of the examples further to include a simple http server:

``` JavaScript
var http = require('http');
var PosixMQ = require('pmq');

var server = http.createServer();

server.on('request', function(req, res) {
    res.writeHead(200, {'Content-Type': 'text/plain'});
    res.write('Hello World!');
    res.end();
});

server.listen(4000);

var readbuf, mq;

mq = new PosixMQ();
mq.on('messages', function() {
    var n;
    while((n = this.shift(readbuf)) !== false) {
        console.log('Received message (' + n + ' bytes): ' + readbuf.toString('utf8', 0, n));
        console.log('Messages left: ' + this.curmsgs);
    }
    this.unlink();
    this.close();
});

mq.open({ name: '/pmqtest', create: true, mode: '0777' });
readbuf = new Buffer(mq.msgsize);
```

After starting the application, open a web browser and point it to http://localhost:4000.  After hitting the refresh button several times (at least 10x), node gives a segmentation fault

Comment out the mq.open line and node does not issue any segmentation faults.

**Comment by mscdex** [comment]
Should be fixed in 39d971c.

**Comment by jkuba** [comment]
That fixed the segmentation fault issue.  However, now if I send a message to the queue in the application above (using the example provided in the readme to fill a queue), I get the following error

```
Received message (1 bytes): 9
Messages left: 9
Received message (1 bytes): c
Messages left: 8
Received message (1 bytes): J
Messages left: 7
Received message (1 bytes): /
Messages left: 6
Received message (1 bytes): )
Messages left: 5
Received message (1 bytes): p
Messages left: 4
Received message (1 bytes): !
Messages left: 3
Received message (1 bytes): N
Messages left: 2
Received message (1 bytes): K
Messages left: 1
Received message (1 bytes): 8
Messages left: 0

/home/ubuntu/pmq_test/pmq_sink.js:24
    this.close();
         ^
Error: no such file or directory
    at PosixMQ.mq.open.name (/home/ubuntu/pmq_test/pmq_sink.js:24:7)
    at PosixMQ.EventEmitter.emit (events.js:92:17)

```

**Comment by jkuba** [comment]
Found the bug in the close() function.  The mqdes variable was being cleared before calling mq_close().  Fix shown below:

``` C
    int close() {
      int r = 0;
      if (mqdes != MQDES_INVALID) {
        uv_poll_stop(mqpollhandle);
        uv_close((uv_handle_t *)mqpollhandle, on_close);
        r = mq_close(mqdes);
        mqdes = MQDES_INVALID;
      }
      return r;
    }
```

**Comment by mscdex** [comment]
Ok, fixed in 943d194.

**Comment by jkuba** [comment]
It works,thanks!
